website/versioned_docs/version-5.x/testing/units.md

```diff
@@ -0,0 +1,109 @@
+
+:::
+
+## `System.import`
+
+Occasionally you will choose to interop with another microfrontend asynchronously by explicitly calling `System.import`. Testing in this scenairo may require mocking both SystemJS and the module you're importing. Additionally because `System.import` returns a promise your tests in that area will need to be asynchronous and wait for promises to resolve.
+
+An example of this can be found in `people` and `planets` applications from `react.microfrontends.app`. The [People application](https://github.com/react-microfrontends/people/blob/master/src/react-mf-people.js#L21) exports a function that resolves with a component. The [Planets Application](https://github.com/react-microfrontends/planets/blob/main/src/planets-page/selected-planet/selected-planet.component.js) imports and uses that component asynchronously with `React.lazy`. Testing this component would necessitate mocking both `SystemJS` and `People`.
+
+## Shared Mocks
+
+If each project mocks every other microfrontend it is possible that the mocks will eventually become out of sync with the actual deployed microfrontend. One way to prevent this is to share mocks so that keeping the mocks in sync only requires one change instead of updating mocks in many different repositories.
+
+## Testing examples
+
+### Jest
+In the above examples I showed how `People` imports a component from `styleguide`. In order to unit test the component in people with Jest you will need to [configure](https://jestjs.io/docs/configuration) jest and mock the `styleguide` MFE. In jest configuration is done via multiple areas.
+1. [Create a jest config file](#jest-config)
+1. [Setup a mocks directory at the root](#mocks-directory)
+1. [Add a setupFile](#setup-file)
+
+#### Jest config
+Jest is configured with a configuration file. Jest supports many options of configuration, below is an example configuration using some of the options. See [more options on Jest's official documentation site](https://jestjs.io/docs/configuration).
+````js
+module.exports = {
+  collectCoverageFrom: ['src/**/*.js'],
+  modulePathIgnorePatterns: ['/cache', '/dist'],
+  transform: {
+    '^.+\\.(j|t)sx?$': 'babel-jest',
+  },
+  setupFilesAfterEnv: ['./jest.setup.js'],
+  moduleNameMapper: {
+    // Note this is only needed if you don't match the module name directly
+    // an alternative would be to place the mock in 
+    // <rootDir>/__mocks__/@react-mf/styleguide.js and it would be autodetected
+    '@react-mf/styleguide': '<rootDir>/__mocks__/styleguide.js',
+  },
+}
+````
+
+#### mocks directory
+Jest will detect folders named `__mocks__` and if the naming convention is exact or the modules have been mapped using `moduleNameMapper` then Jest will use those mocks in place of an import. This structure is essential for other microfrontends where you don't have the code locally. [See more information on jest's official documentation](https://jestjs.io/docs/manual-mocks)
```

Comment:
This isn't true - jest doesn't automatically look for `__mocks__`. You have to specify the folder in the moduleNameMapper

Reply:
https://github.com/react-microfrontends/planets/pull/20

I don't have `moduleNameMapper` specified [here](https://github.com/react-microfrontends/planets/pull/20/files#diff-1e058ca1442e46581b13571fb8d261f9e1f5657e26c96634d4c1072f0f0347f1R1) at all.

Reply:
I'm guessing this is a miscommunication. 

Would something like the following be more clear?

## Mocks Directory

Mocking imports of microfrontends in jest can be done in three ways. 
1. Use `moduleNameMapper` to map mock files to their import name. 
````
// jest.config.js
...8
moduleNameMapper: {
    '@react-mf/api': '<rootDir>/__mocks__/api.js',
  },
````
2. Use jest folder structure convention.
 - If you create a directory at the root of the project and name it `__mocks__` you can then mock your modules by name in that directory. For example if I wanted to mock `@react-mf/api` which has a `/`you would have a directory structure like this:
```
- __mocks__
--- @react-mf
------ api.js
```
3. Mock in jest setup files.

Reply:
Ah I didn't know that - that's a cool feature of jest.